homeassistant/components/subaru/device_tracker.py

```diff
@@ -0,0 +1,92 @@
+"""Support for Subaru device tracker."""
+from __future__ import annotations
+
+from typing import Any
+
+from subarulink.const import LATITUDE, LONGITUDE, TIMESTAMP
+
+from homeassistant.components.device_tracker import SourceType
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from . import get_device_info
+from .const import (
+    DOMAIN,
+    ENTRY_COORDINATOR,
+    ENTRY_VEHICLES,
+    VEHICLE_HAS_REMOTE_SERVICE,
+    VEHICLE_STATUS,
+    VEHICLE_VIN,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Subaru device tracker by config_entry."""
+    entry: dict = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator: DataUpdateCoordinator = entry[ENTRY_COORDINATOR]
+    vehicle_info: dict = entry[ENTRY_VEHICLES]
+    entities: list[SubaruDeviceTracker] = []
+    for vehicle in vehicle_info.values():
+        if vehicle[VEHICLE_HAS_REMOTE_SERVICE]:
+            entities.append(SubaruDeviceTracker(vehicle, coordinator))
+    async_add_entities(entities)
+
+
+class SubaruDeviceTracker(
+    CoordinatorEntity[DataUpdateCoordinator[dict[str, Any]]], TrackerEntity
+):
+    """Class for Subaru device tracker."""
+
+    _attr_icon = "mdi:car"
+    _attr_has_entity_name = True
+    name = None
```

Comment:
Why do we have this?

Reply:
```python
    _attr_icon = "mdi:car"
```
I thought the car icon represented this entity better than the default device tracker icon. The documentation shows this as a recommended shorter form pattern to override the default platform icon.

```python
    _attr_has_entity_name = True
    name = None
```
I'm trying to make it so that the name of the device tracker entity is just the vehicle name, and the entity_id is `device_tracker.<vehicle_name>`. If I don't include both of these lines, the device tracker either fails or is assigned an unfriendly entity name (`subaru <vin> location`) and entity_id (`device_tracker.subaru_<vin>_location`). 

Please let me know if there is a better pattern to use to achieve this.

I was trying to follow these guides: 
https://developers.home-assistant.io/docs/core/entity#entity-naming
https://developers.home-assistant.io/blog/2023/06/28/entity-name-changes/

Reply:
`name` needs to be `_attr_name`